presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1401,6 +1410,45 @@ static protocol::Duration getUptime(
   return protocol::Duration(seconds, protocol::TimeUnit::SECONDS);
 }
 
+void PrestoServer::convertToVeloxPlan(
+    proxygen::HTTPMessage* message,
+    proxygen::ResponseHandler* downstream,
+    const std::vector<std::unique_ptr<folly::IOBuf>>& body) {
+  std::string error;
+  try {
+    auto headers = message->getHeaders();
+
+    std::ostringstream oss;
+    for (auto& buf : body) {
+      oss << std::string((const char*)buf->data(), buf->length());
+    }
+    std::string planFragmentJson = oss.str();
+    protocol::PlanFragment planFragment = json::parse(planFragmentJson);
+
+    auto queryCtx = core::QueryCtx::create();
+    VeloxInteractiveQueryPlanConverter converter(queryCtx.get(), pool_.get());
+
+    // Create static taskId and empty TableWriteInfo needed for plan conversion
+    protocol::TaskId taskId = "velox-plan-conversion.0.0.0";
+    auto tableWriteInfo = std::make_shared<protocol::TableWriteInfo>();
```

Comment:
TODO: this causes a problem for create table, need to address

Reply:
This doesn't seem to be a problem anymore, correct me if I'm missing anything here

Reply:
I discussed with @tdcmeehan and since `TableWriteInfo` is not built until after the planning stage, we need to have an alternative to handle fragments with a `TableWriterNode`. Some alternatives discussed:

1. Adding a field to the TableWriteInfo to indicate that the converter should proceed and skip the required fields, like `ExecutionWriterTarget`
2. Make a dummy TableWriteInfo with the fields populated enough to pass the conversion
3. Skip the TableWriteNode portion of the fragment, the source node can continue with the conversion

All of these options would result in an incomplete conversion of a fragment with a TableWriteNode, but since that is not available at the time the plan validation happens, there isn't much we can do. Here I implemented option (3) which is the least intrusive to the plan conversion.